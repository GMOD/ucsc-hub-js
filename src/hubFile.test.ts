// @ts-nocheck
import fs from 'fs'
import HubFile from './hubFile'

describe('HubFile', () => {
  it('reads a basic hub.txt file', () => {
    const input = fs.readFileSync('test/basic.hub.txt', 'utf8')
    const hubFile = new HubFile(input)
    expect(hubFile).toMatchSnapshot()
    expect(hubFile.toString()).toEqual(input)
  })

  it("throws if the file doesn't start with a hub entry", () =>
    expect(
      () => new HubFile('trackHub UCSCHub\nshortLabel UCSC Hub\n'),
    ).toThrow(/file must begin with a line like/))

  it('throws if a hub is missing a required field', () =>
    expect(() => new HubFile('hub UCSCHub\nshortLabel UCSC Hub\n')).toThrow(
      /file is missing required entr(y|ies):/,
    ))

  it('throws if a hub has an invalid field', () =>
    expect(() => new HubFile('hub UCSCHub\ninvalid invalid Label\n')).toThrow(
      /file has invalid entr(y|ies):/,
    ))
})
