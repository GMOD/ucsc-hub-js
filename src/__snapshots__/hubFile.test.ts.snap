// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubFile reads a basic hub.txt file 1`] = `
Map {
  "hub" => "UCSCHub",
  "shortLabel" => "UCSC Hub",
  "longLabel" => "UCSC Genome Informatics Hub for human DNase and RNAseq data",
  "genomesFile" => "genomes.txt",
  "email" => "genome@soe.ucsc.edu",
  "descriptionUrl" => "ucscHub.html",
}
`;
